plugins/dns/custom/conf/ose-dns-custom

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+# Usage
+function usage {
+    echo "Usage: $0 <args>"
+    echo
+    echo "          --action {add|delete}"
+    echo "          --host hostname - FQ Hostname"
+    echo "          --cname         - CNAME to add to host"
+    echo
+    exit 1
+}
+
+# options followed by one colon indicate they have a required argument
+if ! options=$(getopt -o ahc: -l action:,host:,cname: -- "$@")
+then
+   usage
+fi
+
+set -- $options
+
+while [ $# -gt 0 ]
+do
+    case $1 in
+    --action)  action=`eval echo $2` ; shift ;;
+    --host)    osehost=`eval echo $2`   ; shift ;;
+    --cname)   cname=`eval echo $2`  ; shift ;;
+    (--) shift; break;;
+    (-*) echo "$0: error - unrecognized option $1" 1>&2; exit 1;;
+    (*) break;;
+    esac
+    shift
+done
+
+echo
+echo [ose-dns-custom] INFO: Input Param: action = $action
+echo [ose-dns-custom] INFO: Input Param: host = $osehost
+echo [ose-dns-custom] INFO: Input Param: cname = $cname
+
+
+if [ -z $action ] ; then
+    echo
+    echo "   action is a required option with arguments: {add|delete}, exiting."
+    echo
+    usage
+fi
+
+if [ -z $osehost ] && [ "$action" = "add" ] ; then
+    echo
+    echo "   host is a required option when action is add, exiting."
+    echo
+    usage
+fi
+
+if [ -z $cname ] ; then
+    echo
+    echo "   cname is a required option, exiting."
+    echo
+    usage
+fi
+
+
+# Main
+
+# Params for nsupdate
+hostname=127.0.0.1
+priv_key=/etc/openshift/example.com.key
+ttl=60
+zone=example.com
```

Comment:
Should these parameters be read in from a configuration file? Few actual openshift environments have named installed on the same host as a broker, so this plugin won't work for most.

They should either be read in from a configuration file or passed into the script as parameters. Normally, dns plugins establish connection information in the plugin class's `initialize` method. I can see why that wasn't done here, since it would be cumbersome to pass the connection information every time the script is called. If the connection information isn't passed as parameters, they should at least be read in from a configuration file.

One issue with defining the connection parameters in this script is that this file is not marked as a configuration file in the rpm spec. If a newer version of the rpm is installed, it will overwrite the modified file with the one in the newer rpm.